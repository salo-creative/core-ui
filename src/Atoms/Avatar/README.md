# Avatar

Renders an image as a circlular avatar

---

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import Avatar from '@salo/core-ui/Molecules/Avatar';
```

Implement as follows

```javascript
<Avatar
  size='M'
  image='http://i.stack.imgur.com/Dj7eP.jpg'
/>
```

**For full prop types and usage see storybook info/knobs**