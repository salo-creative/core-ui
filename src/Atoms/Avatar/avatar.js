import React from 'react';
import PropTypes from 'prop-types';

// COMPONENTS
import { AvatarWrapper, TextWrapper } from './avatar.styles';

// HELPERS & CONSTANTS
import { colours } from '../../helpers/colours';

const Avatar = (props) => {
  const {
    className,
    colour,
    firstName,
    image,
    lastName,
    onClick,
    size
  } = props;

  const hasAction = typeof onClick === 'function';

  const handleClick = (e) => {
    if (hasAction) {
      onClick({
        e
      });
    }
  };
  
  return (
    <AvatarWrapper
      className={ className }
      size={ size }
      image={ image }
      background={ colour }
      onClick={ handleClick }
      cursor={ hasAction ? 'pointer' : 'auto' }
    >
      <TextWrapper
        colour={ colour }
      >
        { !image ? `${ firstName.charAt(0) }${ lastName ? lastName.charAt(0) : '' }` : '' }
      </TextWrapper>
    </AvatarWrapper>
  );
};

Avatar.defaultProps = {
  className: null,
  image: '',
  firstName: 'Undefined',
  lastName: '',
  colour: colours.blue,
  onClick: null,
  size: 100
};

Avatar.propTypes = {
  image: PropTypes.string,
  className: PropTypes.string,

  // Fallback if image not available
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  colour: PropTypes.string,
  onClick: PropTypes.func,
  size: PropTypes.number
};

export default Avatar;