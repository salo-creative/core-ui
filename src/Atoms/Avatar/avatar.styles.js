import styled from 'styled-components';

export const AvatarWrapper = styled.button`
  font-size: ${ ({ size }) => `${ size }px` };
  width: 1em;
  height: 1em;
  position: relative;
  border: none;
  background: ${ ({ background }) => background };
  cursor: ${ ({ cursor }) => cursor };
  background-image: url('${ ({ image }) => image }');
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
  align-items: center;
  justify-content: center;
  display: flex;
  &:after {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
  }
`;

export const TextWrapper = styled.p`
  font-size: 0.4em;
  color: ${ ({ theme }) => theme.font };
  text-transform: uppercase;
  text-align: center;
  font-weight: bold;
  user-select: none;
  margin: 0;
  mix-blend-mode: difference;
`;