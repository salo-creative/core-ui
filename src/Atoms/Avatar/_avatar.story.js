import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, number } from '@storybook/addon-knobs';
import { RenderWithProps } from '@jamesbliss/storybook-state';

// Load tests
import { withTests } from '@storybook/addon-jest';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { Avatar, colours } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Atoms | Avatar', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['avatar'] });

stories.add(
  'Basic',
  (() => {
    // KNOBS
    const image = text('Image URL', 'http://i.stack.imgur.com/Dj7eP.jpg');
    const size = number('Size', 100);
    const fallbackText = text('Fallback Text', 'Peter');
    const colour = text('Fallback Colour', colours.blue);

    return (
      <Avatar
        size={ size }
        text={ fallbackText }
        image={ image }
        colour={ colour }
        onClick={ () => alert('Avatar clicked') }
      />
    );
  }),
  {
    info: { propTablesExclude: [RenderWithProps] },
    notes: README
  }
);