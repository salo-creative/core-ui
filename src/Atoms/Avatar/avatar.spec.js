import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import Avatar from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <Avatar />
  );
  expect(container.firstChild).toMatchSnapshot();
});