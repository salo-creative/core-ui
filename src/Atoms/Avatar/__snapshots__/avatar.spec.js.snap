// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c0 {
  font-size: 100px;
  width: 1em;
  height: 1em;
  position: relative;
  border: none;
  background: #00ACED;
  cursor: auto;
  background-image: url('');
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0:after {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
}

.c1 {
  font-size: 0.4em;
  color: #333333;
  text-align: center;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 0;
  mix-blend-mode: difference;
}

<button
  class="c0"
  cursor="auto"
  size="100"
>
  <p
    class="c1"
  >
    AF
  </p>
</button>
`;
